src/metkit/api/metkit_c.h

```diff
@@ -0,0 +1,238 @@
+ * @param request Request instance
+ * @param param parameter name in request
+ * @param index index of value to retrieve for param in request
+ * @param value retrieved value
+ * @return int Error code
+ */
+int metkit_marsrequest_value(const metkit_marsrequest_t* request, const char* param, int index, const char** value);
+
+/** Returns values for specific parameter Request object
+ * @param request Request instance
+ * @param param parameter name in request
+ * @param values array of values for param in request
+ * @param numValues number of values for param in request
+ * @return int Error code
+ */
+int metkit_marsrequest_values(const metkit_marsrequest_t* request, const char* param, const char** values[], size_t* numValues);
+
+/** Populates empty Request object by expanding existing request
+ * @param request Request instance to be expanded
+ * @param expandedRequest empty Request instance to be populated
+ * @param inherit if true, populate expanded request with default values
+ * @param strict it true, raise error rather than warning on invalid values
+ * @return int Error code
+ */
+int metkit_marsrequest_expand(const metkit_marsrequest_t* request, metkit_marsrequest_t* expandedRequest, bool inherit = true, bool strict = false);
+
+/** Merges other Request object into existing request
+ * @param request Request instance to contain result of merge
+ * @param otherRequest other Request instance to merge
+ * @return int Error code
+ */
+int metkit_marsrequest_merge(metkit_marsrequest_t* request, const metkit_marsrequest_t* otherRequest);
+
+/* ---------------------------------------------------------------------------------------------------------------------
+ * REQUEST ITERATOR
+ * --- */
+
+/** Deallocates RequestIterator object and associated resources.
+ * @param it RequestIterator instance
+ * @return int Error code
+ */
+int metkit_free_requestiterator(const metkit_requestiterator_t* it);
+
+/** Moves to the next Request element in RequestIterator
+ * @param it RequestIterator instance
+ * @return int Error code
+ */
+int metkit_requestiterator_next(metkit_requestiterator_t* it);
+
+/** Populates empty Requestion object with data from current element in RequestIterator
+ * @param it RequestIterator instance
+ * @param request empty Request instance to populate with data
+ * @return int Error code
+ */
+int metkit_requestiterator_request(metkit_requestiterator_t* it, metkit_marsrequest_t* request);
+
+/* ---------------------------------------------------------------------------------------------------------------------
+ * PARAM ITERATOR
+ * --- */
+
+/** Deallocates ParamIterator object and associated resources.
+ * @param it ParamIterator instance
+ * @return int Error code
+ */
+int metkit_free_paramiterator(const metkit_paramiterator_t* it);
+
+/** Moves to the next string element in ParamIterator
+ * @param it ParamIterator instance
+ * @return int Error code
+ */
+int metkit_paramiterator_next(metkit_paramiterator_t* it);
```

Comment:
I think this would make for a nicer API if we could use it like this:
```
while((res = metkit_paramiterator_next(it))) {
   //...
}
```
Ofc this would only work if the iteration cannot create errors. But return a nullptr on end of iteration makes for a more natural api imo.